src/app.md

```diff
@@ -725,6 +726,81 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-semver-latest]: #ex-semver-latest
+<a name="ex-semver-latest"></a>
+## Find the latest version satisfying given range
+[![semver-badge]][semver] [![cat-config-badge]][cat-config]
+
+Given a list of version &strs, finds the latest [`semver::Version`] that satisfying a given [`semver::VersionReq`] using [`VersionReq::matches`].
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate semver;
+
+use semver::{AlphaNumeric, Numeric, Version, VersionReq};
+use std::vec::IntoIter;
+#
+# error_chain! {
+#     foreign_links {
+#         SemVer(semver::ReqParseError);
+#     }
+# }
+
+fn find_max_matching_version<'a, I>(version_req_str: &str, iterable: I) -> Result<Option<Version>>
+where
+    I: IntoIterator<Item = &'a str>,
+{
+    let vreq = VersionReq::parse(version_req_str)?;
+
+    Ok(
+        iterable
+            .into_iter()
+            .filter_map(|s| Version::parse(s).ok())
+            .filter(|s| vreq.matches(s))
+            .max(),
+    )
+}
+
+fn run() -> Result<()> {
+    assert_eq!(
+        find_max_matching_version("<= 1.0.0", vec!["0.9.0", "1.0.0", "1.0.1"])?,
+        Some(Version {
+            major: 1,
+            minor: 0,
+            patch: 0,
+            pre: vec![],
+            build: vec![],
+        })
+    );
+
+    // Shows Semver precedence for pre-release tags
+    assert_eq!(
+        find_max_matching_version(
+            ">1.2.3-alpha.3",
+            vec![
+                "1.2.3-alpha.3",
+                "1.2.3-alpha.4",
+                "1.2.3-alpha.10",
+                "1.2.3-beta.4",
+                "3.4.5-alpha.9",
+            ]
+        )?,
+        Some(Version {
```

Comment:
I'm not sure if we should not replace this multiline `Some(Version{...})` with just `Some(Version::parse("1.0.0")?)` for brevity in both cases (also reducing number of required imports). But it will not be a disaster if we leave it as it is. What are your thoughts on it @anna-liao , @ludwigpacifici ?